codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -285,14 +285,28 @@ public String stringShape(StringShape shape) {
                         }
                    }
                }()""".formatted(dataSource, dataSource, underlyingType);
+        }
+        else {
+            return """
+                func() (string) {
+                    var s string
+                    for i := dafny.Iterate(%s) ; ; {
+                        val, ok := i()
+                        if !ok {
+                            return []string{s}[0]
```

Comment:
This is only required when converting from val to * type.

Reply:
Done